pkg/storage/chunk/client/alibaba/oss_object_client.go

```diff
@@ -52,11 +54,35 @@ func (cfg *OssConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.Endpoint, prefix+"oss.endpoint", "", "oss Endpoint to connect to.")
 	f.StringVar(&cfg.AccessKeyID, prefix+"oss.access-key-id", "", "alibabacloud Access Key ID")
 	f.StringVar(&cfg.SecretAccessKey, prefix+"oss.secret-access-key", "", "alibabacloud Secret Access Key")
+	f.Int64Var(&cfg.ConnectionTimeoutSec, prefix+"oss.conn-timeout-sec", 30, "Connection timeout in seconds")
+	f.Int64Var(&cfg.ReadWriteTimeoutSec, prefix+"oss.read-write-timeout-sec", 60, "Read/Write timeout in seconds")
+}
+
+func (cfg *OssConfig) Validate() error {
+	if len(cfg.Bucket) == 0 {
```

Comment:
by default loki validates all provider configs, this would fail if someone is using a different provider. CI is failing as a result, can you remove the extra validations?

I would suggest only validating `ReadWriteTimeoutSec` to ensure it is gt 0 as a zero value would result in all reads and writes to fail immediately as a result of timeout. `ConnectionTimeoutSec` can be gt or eq to 0 as it supports indefinite waits

Reply:
Ok! modified it.